sourced/ml/transformers/content2ids.py

```diff
@@ -146,17 +146,3 @@ def process(self, rows: RDD):
             .map(lambda x: Row(
                     token=x,
                     token_split=" ".join(TokenParser(min_split_length=1).split(x))))
-
-
-class ContentExtractor(Transformer):
```

Comment:
you change it to
```python
-        .link(ContentExtractor()) \
+        .link(LanguageSelector(languages=args.languages)) \
```
here: https://github.com/src-d/ml/pull/230/files#diff-0a78c1a1a739541ac75c64d15c310dbeL15
and it is not the same.

We filter here only by the rule `"lang is not null"` and in `LanguageSelector` we have different. 
Also, we filter by content. 
Please check. 
You can
1.  leave it as is or 
2. modify `LanguageSelector` and maybe add some additional transformers.

Reply:
1. Regarding content, your are right I forgot to add the filtering to the ContentExtractor
2. Regarding language, after speaking with Waren he told me that yes, we could not filter language that way. Whenever Babelfish supports as much languages as Pygments it will not be a problem, however until then we should keep the "lang is not null" rule. I think the most logical is to do the following:
- make default language param to None
- if None, use rule "lang is not null"
- else do it that way
gonna push change, tell me what u think

Reply:
> if language is None, use rule "lang is not null"
it is counterintuitive a bit. You can pass it as a string "lang is not null" instead of None.

Reply:
okay, but I dont want to hardcode a test `if self.languages == lang is not null`, and using `lang_filter = classified.lang == "not null"` or `lang_filter = classified.lang == "lang is not null"` will not work. I think in this case it is best to add a boolean `blacklist` that defaults to False, but if set to True will exclude by language, e.g. for languages = [null, Python, Java]: 
- `blacklist` is False: `lang_filter = lang == null OR lang == Python OR lang == Java`
- `blacklist` is True: `lang_filter = lang != null AND lang != Python AND lang == Java`
That way we will be able to exclude languages in an easier way then by passing a long list of languages to keep

Reply:
ok.